Content.Client/GameObjects/Components/Photography/PhotoCameraComponent.cs

```diff
@@ -0,0 +1,204 @@
+                {
+                    message.AddMarkup(Loc.GetString("[color=green]On[/color]\n"));
+                }
+                else
+                {
+                    message.AddMarkup(Loc.GetString("[color=red]Off[/color]\n"));
+                }
+
+                message.AddMarkup(Loc.GetString("Film: [color={0}]{1}/{2}[/color], ",
+                    _parent.Film <= 0 ? "red" : "white", _parent.Film, _parent.FilmMax));
+                message.AddMarkup(Loc.GetString("Radius: [color=white]{0}x{0}[/color]", _parent.Radius * 2));
+
+                _label.SetMessage(message);
+            }
+        }
+
+        public override void HandleNetworkMessage(ComponentMessage message, INetChannel netChannel, ICommonSession session = null)
+        {
+            base.HandleNetworkMessage(message, netChannel, session);
+
+            switch (message)
+            {
+                case SuicideSelfieMessage selfie:
+                    var viewport = _eyeManager.GetWorldViewport();
+                    var center = _eyeManager.WorldToScreen(_playerManager.LocalPlayer.ControlledEntity.Transform.GridPosition);
+                    TryTakePhoto(selfie.Who, new Vector2(center.X, center.Y), true);
+                    break;
+            }
+        }
+
+        public async void TryTakePhoto(EntityUid author, Vector2 photoCenter, bool suicide = false)
+        {
+            if (!CameraOn)
+            {
+                _notifyManager.PopupMessageCursor(_playerManager.LocalPlayer.ControlledEntity, Loc.GetString("Turn the {0} on first!", Owner.Name));
+                return;
+            }
+
+            if(Film <= 0)
+            {
+                _notifyManager.PopupMessageCursor(_playerManager.LocalPlayer.ControlledEntity, Loc.GetString("No film!"));
+                return;
+            }
+
+            //Play sounds
+            SendNetworkMessage(new TakingPhotoMessage());
+
+            //Take a screenshot before the UI, and then crop it to the photo radius
+            var screenshot = await _clyde.ScreenshotAsync(ScreenshotType.BeforeUI);
+            var cropDimensions = EyeManager.PixelsPerMeter * (Radius * 4);
+
+            //We'll try and center, but otherwise we'll shift the box so it doesn't go outside the viewport
+            var cropX = (int)Math.Clamp(Math.Floor(photoCenter.X - cropDimensions / 2), 0, screenshot.Width - cropDimensions);
+            var cropY = (int)Math.Clamp(Math.Floor(photoCenter.Y - cropDimensions / 2), 0, screenshot.Height - cropDimensions);
+
+            Logger.InfoS("photo", $"cropX:{cropX}, cropY:{cropY}, w:{screenshot.Width}, h:{screenshot.Height}");
+
+            var success = false;
+            var filename = $"MY_PHOTO";
+            using (screenshot)
+            {
+                screenshot.Mutate(
+                    i => i.Crop(new Rectangle(cropX, cropY, cropDimensions, cropDimensions))
+                );
+
+                for (var i = 0; i < 5; i++)
```

Comment:
Does this only support 5 photo files?

Reply:
This is 5 attempts to save the same file.
Exact same precautions PJ took with screenshots.
In practice I've only noticed it get as bad as the third attempt to save it (dash 2).

Since this is just the photo you last took, there's no need to keep it around forever, so they get overwritten if you take another photo.

(Ideally on client shutdown or somewhere we'd then delete all of these (probably need to make a /Photos directory like there's a /Screenshots one, just for organisation))